Source/WebCore/dom/InternalObserverLast.cpp

```diff
@@ -53,45 +54,44 @@ class InternalObserverLast final : public InternalObserver {
 private:
     void next(JSC::JSValue value) final
     {
-        m_lastValue = value;
+        m_lastValue.setWeakly(value);
     }
 
     void error(JSC::JSValue value) final
     {
-        Ref { m_promise }->reject<IDLAny>(value);
+        protectedPromise()->reject<IDLAny>(value);
     }
 
     void complete() final
     {
         InternalObserver::complete();
 
-        auto lastValue = std::exchange(m_lastValue, std::nullopt);
+        if (UNLIKELY(!m_lastValue))
+            return protectedPromise()->reject(Exception { ExceptionCode::RangeError, "No values in Observable"_s });
 
-        if (UNLIKELY(!lastValue))
-            return Ref { m_promise }->reject(Exception { ExceptionCode::RangeError, "No values in Observable"_s });
-
-        Ref { m_promise }->resolve<IDLAny>(*lastValue);
+        protectedPromise()->resolve<IDLAny>(m_lastValue.getValue());
```

Comment:
I don't think it matters, given once we `Complete`, the `visitAdditionalChildren` will stop being ran, as the m_active flag gets set to false upstream.

But is it worth calling `.clear` on the value?

```suggestion
        protectedPromise()->resolve<IDLAny>(m_lastValue.getValue());
        
        m_lastValue.clear();
```